packages/webapp/src/containers/Animals/AddAnimals/AddAnimalBasics/useUpdateBasics.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useLayoutEffect } from 'react';
+import { useFormContext, UseFieldArrayReplace } from 'react-hook-form';
+import { useAnimalOptions } from '../useAnimalOptions';
+import {
+  AnimalBasicsFormFields,
+  AnimalDetailsFormFields,
+  BasicsFields,
+  DetailsFields,
+} from '../types';
+import { AddAnimalsFormFields } from '../types';
+import { STEPS } from '..';
+import { Details as SexDetailsType } from '../../../../components/Form/SexDetails/SexDetailsPopover';
+
+export const useUpdateBasics = (
+  fields: AnimalBasicsFormFields[],
+  replace: UseFieldArrayReplace<AddAnimalsFormFields, 'basics'>,
+) => {
+  const { getValues } = useFormContext<AddAnimalsFormFields>();
+  const { sexDetailsOptions } = useAnimalOptions('sexDetails');
```

Comment:
Would it make sense to pass this in from parent -- the parent calls this twice otherwise.